### PR TITLE
Fixed the time step sanity check with DualController.

### PR DESCRIPTION
**Pull request by nloppi**

### CODE REVIEW

File: pyfr/integrators/dual/phys/base.py [hunk 1]
```diff
@@ -44,3 +46,11 @@ def soln(self):
             )
 
         return self._curr_soln
+
+    def call_plugin_dt(self, dt):
+        rem = math.fmod(dt, self._dt)
+        tol = 5.0*self.dtmin
+        if rem > tol and (self._dt - rem) > tol:
+            raise ValueError('Output times must be multiplies of dt')
```
Comment (FreddieWitherden) [inline pyfr/integrators/dual/phys/base.py]: Plugin times.

Comment (pv101) [reply pyfr/integrators/dual/phys/base.py]: multiplies -> multiples

Comment (nloppi) [reply pyfr/integrators/dual/phys/base.py]: Plugin output times?

File: pyfr/integrators/dual/phys/base.py [hunk 2]
```diff
@@ -44,3 +46,11 @@ def soln(self):
             )
 
         return self._curr_soln
+
+    def call_plugin_dt(self, dt):
+        rem = math.fmod(dt, self._dt)
+        tol = 5.0*self.dtmin
+        if rem > tol and (self._dt - rem) > tol:
+            raise ValueError('Plugin output times must be multiples of dt')
```
Comment (FreddieWitherden) [inline pyfr/integrators/dual/phys/base.py]: Maybe call is better than output?

**Comment by FreddieWitherden** [comment]
Okay!  Final sanity check is to convince ourselves that self._dt is always set when call_plugin_dt is invoked.

**Comment by nloppi** [comment]
> Okay! Final sanity check is to convince ourselves that self._dt is always set when call_plugin_dt is invoked.

Yes.

self._dt is set in the BaseIntegrator constructor. The super().__init__(*) in BaseDualIntegrator is called before self.completed_step_handlers = proxylist(self._get_plugins()). You can see it in

class BaseDualIntegrator(BaseIntegrator):
    formulation = 'dual'

    def __init__(self, backend, systemcls, rallocs, mesh, initsoln, cfg):
        super().__init__(backend, rallocs, mesh, initsoln, cfg)

        # Get the pseudo-integrator
        self.pseudointegrator = get_pseudo_integrator(
            backend, systemcls, rallocs, mesh,
            initsoln, cfg, self._stepper_coeffs, self._dt
        )

        # Event handlers for advance_to
        self.completed_step_handlers = proxylist(self._get_plugins())

        # Delete the memory-intensive elements map from the system
        del self.system.ele_map

**Pull request merged**